milli/src/search/new/bucket_sort.rs

```diff
@@ -143,10 +174,12 @@ pub fn bucket_sort<'ctx, Q: RankingRuleQueryTrait>(
         ranking_rule_universes[cur_ranking_rule_index] -= &next_bucket.candidates;
 
         if cur_ranking_rule_index == ranking_rules_len - 1
-            || next_bucket.candidates.len() <= 1
+            || (scoring_strategy == ScoringStrategy::Skip && next_bucket.candidates.len() <= 1)
             || cur_offset + (next_bucket.candidates.len() as usize) < from
         {
             maybe_add_to_results!(next_bucket.candidates);
+            // FIXME: use index based logic like all the other rules so that you don't have to maintain the pop/push?
+            ranking_rule_scores.pop();
```

Comment:
Can we refer this change in an issue or can we do it right now?

Reply:
unfortunately, it is hard to do due to score details not having a meaningful "default" value. The options I see:

1. keep it this way (it works, but is inconsistent)
2. Make the vector a vector of `Option<ScoreDetails>` so we have a default value. I find it sad since everytime the option will be actually accessed it will have to be a Some that we will subsequently unwrap.
3. Use a default `ScoreDetails` value. They all have date, so which one should we use?

Do you see another option?

Reply:
Let's keep it this way and remove the comment :)

Reply:
done